### DISCUSSION TITLE
combine colored step files

### DISCUSSION THREAD

**Discussion by RBSPhoenix** [CadQuery/cadquery#1454] [Q&A]
Hello, I want to combine step files while keeping the colors of the original files.

In my example, I have a simple step file named pcb.stp, which is basically a green box. I also have a chip package called SOT23-6.stp where the main part is black and the metallic pins are silver/gray. Now I want to place the chip on the pcb and export the assembly as a new step file.

The code I am running in CQ-editor is this:

> import cadquery as cq
> 
> pcb = cq.importers.importStep("C:/temp/PCB_160x100.stp")
> ic1 = cq.importers.importStep("C:/temp/SOT23-6.stp").translate((10,10,0))
> 
> assy = cq.Assembly()
> assy.add(pcb, name="pcb")
> assy.add(ic1, name="ic1")
> 
> assy.save("C:/temp/ass_PCB.step")

but after I open the file, everything is black/dark gray.

![combine](https://github.com/CadQuery/cadquery/assets/119939989/dff5016f-8971-4dc6-b20a-b717030da22a)

Is there any way to change the importers settings? When I google it, I saw a discussion for FreeCAD where someone had the same issue with FreeCAD and he was told to change some settings:

https://forum.freecad.org/viewtopic.php?t=50308

Or is there just no way I can do that with CadQuery? I think its an issue of the importer, because CQ-editor shows me the two parts in orange. I can change the color in the add() command. So I can change the pcb to green. But the chip package has two different colors and I don't know how to achieve my goal to simply add a couple of step files, move them around and then save the assembly as another, combined step file.

Any ideas anyone?

Thanks and have a nice day
Michael

**Comment by jmwright** [comment]
CadQuery's [importStep](https://github.com/CadQuery/cadquery/blob/6869dc625bf29e098204fcd71bbe1025fd553644/cadquery/occ_impl/importers/__init__.py#L44) method uses [STEPControl_Reader](https://dev.opencascade.org/doc/refman/html/class_s_t_e_p_control___reader.html), and I don't think it supports colors. You could drop down to the OCP layer and try to use [STEPCAFControl_Reader](https://dev.opencascade.org/doc/refman/html/class_s_t_e_p_c_a_f_control___reader.html) with the color mode set, but I have not tried to do that so I do not know how/if it will work.
